internal/util/connection.go

```diff
@@ -28,13 +30,23 @@ func ReadFromNetworkConnection(connection net.Conn) ([]byte, error) {
 		return nil, fmt.Errorf("Message content is too large to read.")
 	}
 
-	jsonBuffer := make([]byte, size)
-	_, err = connection.Read(jsonBuffer)
-	if err != nil {
-		return nil, err
+	log.Trace("Reading the following bytes from a connection...", log.NewAttr("expected bytes", size))
+
+	buffer := make([]byte, size)
+	var numBytesRead uint64
+	for numBytesRead = 0; numBytesRead < size; {
```

Comment:
Comment on what you are doing here and why it is necessary.
(Doesn't need to be anything fancy, just 1-2 sentences.)